File: frontend/src/types/team.ts
export type MemberType = "Aparat" | "Profburo" | "Viddil";

export const APARAT_TYPE  = 0;
export const PROFBURO_HEAD_TYPE = 1;
export const VIDDIL_HEAD_TYPE  = 2;

export const MEMBER_TYPE_OPTIONS = [
  { id: APARAT_TYPE, label: "Член Президії", filterLabel: "Члени Президії" },
  { id: PROFBURO_HEAD_TYPE, label: "Голова Профбюро Студентів", filterLabel: "Голови Профбюро Студентів" },
  { id: VIDDIL_HEAD_TYPE, label: "Голова Відділу", filterLabel: "Голови Відділів" },
] as const;

export const MEMBER_TYPE_LABELS: Record<number, string> = Object.fromEntries(
  MEMBER_TYPE_OPTIONS.map(({ id, label }) => [id, label]),
);

export function getMemberTypeLabel(type: number): string {
  return MEMBER_TYPE_LABELS[type] ?? String(type);
}

export interface TeamFormData {
  name: string;
  position: string;
  type: number;
  orderInd: number;
  isTemporary: boolean;
  email?: string;
  imageUrl?: string;
}

export interface TeamMember {
  id: number;
  name: string;
  position: string;
  type: number;
  typeName?: string;
  orderInd: number;
  isTemporary: boolean;
  createdAt: string;
  email?: string;
  imageUrl?: string;
  isChoosed: boolean;
}

File: backend/Backend/Models/Team.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel;
using ProfkomBackend.Utils;

namespace ProfkomBackend.Models
{
    public enum MemberType
    {
        Aparat,
        Profburo,
        Viddil
    }

    public class Team
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(FieldLimits.PersonName)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(FieldLimits.Position)]
        public string Position { get; set; } = string.Empty;

        [Required]
        public MemberType Type { get; set; }

        [NotMapped]
        public string TypeName => MemberTypeNames.Get(Type);

        public string? ImageUrl { get; set; }
        [MaxLength(FieldLimits.Email)]
        [RegularExpression(FieldLimits.EmailPattern, ErrorMessage = FieldLimits.EmailFormatMessage)]
        public string? Email { get; set; }

        [Required]
        public int OrderInd { get; set; }           // number у фронті

        [Required]
        public bool IsTemporary { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow; // created_at

        public DateTime UpdatedAt { get; set; }

        public bool IsChoosed { get; set; } = false;
    }
}
